recipe/meta.yaml

```diff
@@ -1,120 +1,159 @@
 {% set version = "0.17.0" %}
-{% set filename = "apache-arrow-" + version + ".tar.gz" %}
-{% set checksum = "2c46b4c3e1f88aad510214e633a6f4ce459708f3db78cd0daf549a135cbe8e6d" %}
+{% set number = "2" %}
+{% set cuda_enabled = cuda_compiler_version is not undefined and cuda_compiler_version == '9.2' %}
+{% set build_ext_version = "1.0.0" %}
+{% set build_ext = "cuda" if cuda_enabled else "cpu" %}
 
 package:
-  name: arrow-cpp
+  name: arrow-cpp-ext
   version: {{ version }}
 
 source:
-  fn: {{ filename }}
-  url: https://dist.apache.org/repos/dist/release/arrow/arrow-{{ version }}/{{ filename }}
-  sha256: {{ checksum }}
+  url: https://dist.apache.org/repos/dist/release/arrow/arrow-{{ version }}/apache-arrow-{{ version }}.tar.gz
+  sha256: 2c46b4c3e1f88aad510214e633a6f4ce459708f3db78cd0daf549a135cbe8e6d
 
 build:
-  number: 0
-  skip: true  # [win and vc<14]
+  number: {{ number }}
+  skip: true  # [cuda_compiler_version not in (undefined, "None", "9.2")]
   run_exports:
     - {{ pin_subpackage("arrow-cpp", max_pin="x.x.x") }}
 
-requirements:
-  build:
-    - cmake 3.16.*
-    - autoconf  # [unix]
-    - ninja
-    - make  # [unix]
-    - {{ compiler('c') }}
-    - {{ compiler('cxx') }}
-  host:
-    - aws-sdk-cpp
-    - boost-cpp
-    - brotli
-    - bzip2
-    - c-ares
-    - gflags
-    - glog
-    - grpc-cpp
-    - libprotobuf
-    - clangdev 8.*
-    - llvmdev 8.*
-    - lz4-c
-    - numpy 1.16.*
-    - orc  # [unix]
-    - python
-    - rapidjson
-    - re2
-    - snappy
-    - thrift-cpp >=0.11
-    - zlib
-    - zstd
+outputs:
+  - name: arrow-cpp-proc
+    version: {{ build_ext_version }}
+    build:
+      number: 0
+      string: "{{ build_ext }}"
+    test:
+      commands:
+        - exit 0
+    about:
+      home: http://github.com/apache/arrow
+      license: Apache-2.0
+      license_file:
+        - LICENSE.txt
+      summary: 'A meta-package to select Arrow build variant'
 
-  run:
-    - {{ pin_compatible('numpy', lower_bound='1.16') }}
-    - aws-sdk-cpp
-    - boost-cpp
-    - brotli
-    - c-ares
-    - gflags
-    - glog
-    - grpc-cpp
-    - lz4-c
-    - python
-    - re2
-    - snappy
-    - zlib
-    - zstd
+  - name: arrow-cpp
+    script: build-arrow.sh  # [not win]
+    script: bld-arrow.bat   # [win]
+    version: {{ version }}
+    build:
+      number: {{ number }}
+      string: "py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ number }}_{{ build_ext }}"
+      skip: true  # [cuda_compiler_version not in (undefined, "None", "9.2")]
```

Comment:
```suggestion
```

I think it's worth trying to see whether cuda 10.x builds successfully. 9.2. cannot make use of the `compute_75` instruction set, for example.

Reply:
Following this suggestion would need a few more adaptations (using `# [cuda_compiler_version is not undefined and cuda_compiler_version != "None"]` as a selector), plus a rerender

Reply:
That's not necessary. Please read the discussion in the PR that added CUDA support.

Reply:
OK, thanks for the info. To record the outcome (to the best of my understanding): since arrow only uses the cuda driver (and not the toolkit), building for one cuda version is compatible with all.